src/main/java/duke/commands/TasksOnCommand.java

```diff
@@ -0,0 +1,57 @@
+package duke.commands;
+
+import java.time.LocalDate;
+
+import duke.tasklist.TaskList;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.ui.Ui;
+
+/**
+ * The TasksOnCommand class is a Command that returns the tasks which have deadlines/end dates on the input date
+ */
+public class TasksOnCommand extends Command {
+
+    private LocalDate date;
+
+    /**
+     * Public constructor which is used to initialise the input date
+     *
+     * @param date input date which is used to retrieve the tasks
+     */
+    public TasksOnCommand(LocalDate date) {
+        this.date = date;
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Checks whether the list contains a deadline or event with the input date as a deadline/end date and
+     * appends it to a temporary list
+     *
+     * @param taskList the list of tasks upon which the operations need to be performed
+     * @param ui the ui in which the result message of the command execution is displayed to the user
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui) {
+        TaskList list = new TaskList();
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            if (task instanceof Deadline) {
+                if (((Deadline) task).getDate().equals(date)) {
+                    list.add(task);
+                }
+            }
+            if (task instanceof Event) {
+                if (((Event) task).getDate().equals(date)) {
+                    list.add(task);
+                }
+            }
```

Comment:
Since a task is either a Deadline or Event or any other classes that are subclass of Task, it won't be a Deadline or Event at the same time. Hence, I feel that using "if" and "else if" is better than "if and if" since you only need to check if a task is an instance of Event only if it is not an instance of Deadline.
```suggestion
            if (task instanceof Deadline) {
                if (((Deadline) task).getDate().equals(date)) {
                    list.add(task);
                }
            } else if (task instanceof Event) {
                if (((Event) task).getDate().equals(date)) {
                    list.add(task);
                }
            }
```